### PR TITLE
Enforces explicit hostname passing and https

### PR DESCRIPTION
**Pull request by victorfonsec4**

### CODE REVIEW

File: spec/zeus/api_client_spec.rb [hunk 1]
```diff
@@ -19,7 +19,8 @@
 
   let (:zeus_client) do
     Zeus::APIClient.new({
-      access_token: 'fake_access_token'
+      access_token: 'fake_access_token',
+      endpoint: 'luke.skywalker.com'
```
Comment (travelist) [inline spec/zeus/api_client_spec.rb:23]: @victorfonsec4 Thank you for the PR.

For the test, would u mind to add test code something like following?:
- [x] URL starting with `http://`
- [x] URL without protocol (`http://`, `https://`)

``` ruby
# I haven't checked this code, so just an example.
describe '#initialize' do
    context 'endpoint starts with http' do
      it 'replace http to https' do
        fake_zeus_client = Zeus::APIClient.new({
          access_token: 'fake_access_token',
          endpoint: 'http://fake-ciscozeus.io'
        })
        expect(fake_zeus_client.instance_variable_get(:@endpoint)).to eq('https://fake-ciscozeus.io')
      end
    end
    context 'endpoint without protocol' do
      it 'add https to the start of the url' do
        fake_zeus_client = Zeus::APIClient.new({
          access_token: 'fake_access_token',
          endpoint: 'fake-ciscozeus.io'
        })
        expect(fake_zeus_client.instance_variable_get(:@endpoint)).to eq('https://fake-ciscozeus.io')
      end
    end
  end
```

BTW, endpoint URL(`@endpoint`) should have default value:
- Endpoint URLs(currently we are using a just 1 EP though..) should be abstracted, so developer does not concerning each API endpoints, instead they can focus on the functionality.
  - `ruby-zeusclient` users does NOT know each API endpoints in general
  - Endpoint URL dose not change mostly.

We can see some other libraries which have default URLs:
- https://github.com/google/google-api-ruby-client
- https://github.com/sendgrid/sendgrid-ruby

(These libraries are just I quickly searched.)
- [ ] Test code (`#initialize`) without `endpoint`

Thanks!

**Comment by travelist** [comment]
@victorfonsec4 

Sorry for late response. I've just reviewed this PR. 
If you dont mind, please write comment with mentioning `@travelist` when it's ready for the review, so that I can easy to aware of. 

Anyway, thanks for the PRs:)

**Comment by travelist** [comment]
@victorfonsec4 
I just talked with @johnugeorge about the default API endpoints. There seems to be no default EP (it's kind of dynamic EPs), so we can skip it this time.

Meanwhile, it is worth to implementing the API which returns the dynamic EPs(URLs) for given _access_token_ (also, we can check the token at this time). Then, we can modify this client to set EP using _access_token_ when initialize the instance. (Client users do not need to care about Endpoint URLs so..)

Please let me know if there are any questions.

Thanks!

**Comment by victorfonsec4** [comment]
@travelist
Added the tests, basically just copy pasted urs and added an extra one

**Review by travelist — APPROVED** [review]
Nice, Thanks!:) Plz merge it!

**Pull request merged**